### ISSUE TITLE
LoopingCall should be resetable

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5114]
|<img alt="chris-'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| chris- reported|
|-|-|
|Trac ID|trac#5114|
|Type|enhancement|
|Created|2011-05-19 15:56:05Z|
|Branch|https://github.com/twisted/twisted/tree/reset-loop-5114|

It would be a good thing if the LoopingCall would be "resetable", meaning it should be easy to stop the currently scheduled call and reschedule it with the currently set interval, providing a nicer alternative to stopping the call and starting a new LoopingCall with the exact same settings.
The following patch implements this.

Attachments:

* [reset_loop.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ff4/ff46fb9456929dcae605bc34e61129cc8c9b98ed/5a873d3c4e67cf2b18dabf4b18273cd55344cad7.patch) (1570 bytes) - added by chris- on 2011-05-19 15:56:48Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5114 5114
type__enhancement enhancement
reporter__chris_ chris-
priority__normal normal
milestone__ 
branch__branches_reset_loop_5114 branches/reset-loop-5114
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__patch patch
time__1305820565000000 1305820565000000
changetime__1474197555542286 1474197555542286
version__None None
owner__therve therve
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Adding missing 'review' keyword (see ReviewProcess).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [31998]) Branching to 'reset-loop-5114'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [32000]) Merge reset-loop-5114

Author: chris
Reviewer: therve
Fixes: [#5114](https://github.com/twisted/twisted/issues/5114)

Add a reset method to LoopingCall to be able to stop it and reschedule at once.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
